src/components/App.js

```diff
@@ -4,13 +4,32 @@ import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import calculate from '../logic/calculate';
 
-function App() {
-  return (
-    <>
-      <Display calculate={calculate} />
-      <ButtonPanel />
-    </>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: '0',
+      next: '',
+      operation: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { total, next, operation } = calculate(this.state, buttonName);
+    this.setState({ total, next, operation });
+  }
+
+  render() {
+    const { total, next, operation } = this.state;
+    const output = operation === '=' ? total : `${total} ${operation} ${next}`;
+    return (
+      <>
+        <Display result={output} />
```

Comment:
- [ ] it passes the `total` or `next` value from state to `Display` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passToatalToDisplay.png)). So the conditional statement should be similar to this:

<Display result={this.state.next || this.state.total || '0'} />